addons/web/static/src/js/core/owl_popover.js

```diff
@@ -0,0 +1,272 @@
+odoo.define('web.OwlPopover', function() {
+    'use strict';
+
+    const { Component, hooks, misc, QWeb } = owl;
+    const { Portal } = misc;
+    const { useExternalListener, useRef, useState } = hooks;
+
+    /**
+     * Popover (owl version)
+     *
+     * Represents a bootstrap-style popover handled with pure JS. The popover
+     * will be visually bound to its `target` using an arrow-like '::before'
+     * CSS pseudo-element.
+     * @extends Component
+     **/
+    class Popover extends Component {
+        /**
+         * @param {Object} [props]
```

Comment:
props isn't optional

Reply:
i actually didn't know the `[]` meant 'optional' in jsdoc! thx